packages/core/src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -201,6 +202,50 @@ export class HttpResourceFetcher implements ResourceFetcher {
     }
 }
 
+export class RetryableResourceFetcher extends HttpResourceFetcher {
```

Comment:
I think this should probably be done in a different way. Maybe the httpResourceFetcher can have retries in it? I think we can re-visit this after this PR